packages/qdrant-loader/src/qdrant_loader/core/project_manager.py

```
@@ -263,6 +263,29 @@ async def _update_project_sources(
+                        # Drop it so the next run starts fresh instead of
+                        # silently resuming mid-page against a different query.
+                        if current_source_config_hash is not None:
+                            from qdrant_loader.core.state.checkpoint_manager import (
+                                CheckpointManager,
+                            )
+
+                            try:
+                                await CheckpointManager(session).clear_checkpoint(
+                                    project_id, source_type, source_name
+                                )
+                            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)